integrations/gitlab/gitlab_integration/gitlab_service.py

```diff
@@ -650,26 +685,37 @@ async def get_and_parse_single_file(
             )
             return None
 
-    async def get_all_files_in_project(
+    async def get_paginated_files_in_project(
         self, project: Project, path: str
     ) -> typing.AsyncIterator[List[dict[str, Any]]]:
         branch = project.default_branch
         try:
-            file_paths = await self._get_file_paths(project, path, branch, True)
+            tasks: List[Any] = []
             logger.info(
-                f"Found {len(file_paths)} files in project {project.path_with_namespace} files: {file_paths}"
+                f"Getting files in project {project.path_with_namespace} based on pattern {path}"
             )
-            files = []
-            tasks = []
-            for file_path in file_paths:
-                tasks.append(self.get_and_parse_single_file(project, file_path, branch))
-
-                if len(tasks) == PROJECT_FILES_BATCH_SIZE:
-                    results = await asyncio.gather(*tasks)
-                    files.extend([file_data for file_data in results if file_data])
-                    yield files
+            async for file_paths_page in self.get_paginated_file_paths(
+                project, path, branch, True
+            ):
+                logger.info(
+                    f"Found {len(file_paths_page)} files in project {project.path_with_namespace} files: {file_paths_page}"
+                )
```

Comment:
not sure I understand the end of the log, of `files: {file_paths_page}`

Reply:
These are the files that we managed to fetch